data-reporting-service/person-service/src/test/java/gov/cdc/etldatapipeline/person/PersonServiceTest.java

```diff
@@ -127,71 +133,36 @@ public void processProviderElasticSearchData() {
-            String expectedKeyFilePath) {
-        PersonService ks = getKafkaStreamService();
-        StreamsBuilder streamsBuilder = new StreamsBuilder();
-        ks.processMessage(streamsBuilder);
-        Topology topology = streamsBuilder.build();
-        try (TopologyTestDriver topologyTestDriver = new TopologyTestDriver(topology, new Properties())) {
-            TestInputTopic<String, String> inputTopic = topologyTestDriver
-                    .createInputTopic(inputTopicName, new StringSerializer(), new StringSerializer());
-
-            TestOutputTopic<String, String> outputTopic = topologyTestDriver
-                    .createOutputTopic(outputTopicName, new StringDeserializer(), new StringDeserializer());
-            inputTopic.pipeInput("10000001", incomingChangeData);
-            List<KeyValue<String, String>> actualData = outputTopic.readKeyValuesToList();
-            Assertions.assertNotNull(actualData);
-
-            //Validate the Provider Payload
-            TypeReference<DataEnvelope> dataEnvelopeTypeReference = new TypeReference<>() {
-            };
-
-            DataEnvelope<DataEnvelope> actualValue
-                    = objectMapper.readValue(actualData.get(0).value, dataEnvelopeTypeReference);
-            Assertions.assertEquals(
-                    objectMapper.readValue(readFileData(expectedValueFilePath), dataEnvelopeTypeReference),
-                    actualValue);
-
-            //Validate the Patient Key
-            DataEnvelope<DataEnvelope> actualKey
-                    = objectMapper.readValue(actualData.get(0).key, dataEnvelopeTypeReference);
-            //Construct expected Patient Key
-            Assertions.assertEquals(
-                    objectMapper.readValue(readFileData(expectedKeyFilePath), dataEnvelopeTypeReference),
-                    actualKey);
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
-    }
+            String expectedKeyFilePath) throws JsonProcessingException {
+
+        String expectedKey = readFileData(expectedKeyFilePath);
+        String expectedValue = readFileData(expectedValueFilePath);
+
+        System.out.println(expectedKey);
+
+        personService.processMessage(incomingChangeData, inputTopicName);
+
+        ArgumentCaptor<String> topicCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<String> keyCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<String> valueCaptor = ArgumentCaptor.forClass(String.class);
+
+        verify(kafkaTemplate, Mockito.times(2)).send(topicCaptor.capture(), keyCaptor.capture(), valueCaptor.capture());
+
+        JsonNode expectedJsonNode = objectMapper.readTree(expectedKey);
+        JsonNode actualJsonNode = objectMapper.readTree(keyCaptor.getValue());
 
+//        assertEquals(outputTopicName, topicCaptor.getValue());
+        assertEquals(expectedJsonNode, actualJsonNode);
+//        assertEquals(expectedValue, valueCaptor.getValue());
```

Comment:
Looks like we are validating only the key. Are we not validating the event value?

Reply:
Oops! forgot to uncomment it. let me do that in the next rev

Reply:
fixed